plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/RenameElementRefactoringProcessor.java

```diff
@@ -58,10 +66,39 @@ public RefactoringStatus checkFinalConditions(final IProgressMonitor pm, final C
 	@Override
 	public Change createChange(final IProgressMonitor pm) throws CoreException, OperationCanceledException {
 		final CompositeChange change = new CompositeChange(getProcessorName());
-		change.add(new RenameElementChange(elementURI.lastSegment(), elementURI, newName));
+		change.add(new RenameElementChange("Rename Pin in type: " + elementURI.lastSegment(), elementURI, newName));
+		createChildChanges(change);
 		return change;
 	}
 
+	private void createChildChanges(final CompositeChange change) {
+		final TypeEntry typeEntry = TypeLibraryManager.INSTANCE.getTypeEntryForURI(elementURI);
+		final BlockTypeInstanceSearch search = new BlockTypeInstanceSearch(typeEntry);
+		final List<? extends EObject> result = search.performSearch();
+		final var eChild = getChildByURI(typeEntry.getType(), elementURI);
+		String oldName = "";
+		if (eChild instanceof final IInterfaceElement varDecl) {
+			oldName = varDecl.getName();
+		}
+
+		for (final EObject eObject : result) {
+			change.add(new ReconnectPinChange(EcoreUtil.getURI(eObject), FBNetworkElement.class, newName, oldName));
+		}
```

Comment:
Please first check if we are actually renaming an `IInterfaceElement` and avoid performing the search or creating the commands otherwise.

Reply:
I guess we should also only do this when renaming inside an FB type and not, for example, variables in attributes or structs?